packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -587,6 +593,17 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
 export abstract class SharedObject<
 	TEvent extends ISharedObjectEvents = ISharedObjectEvents,
 > extends SharedObjectCore<TEvent> {
+	/**
+	 * Get a factory for SharedMap to register with the data store.
+	 * @returns A factory that creates SharedMaps and loads them from storage.
+	 */
+	public static readonly [FluidStaticEntryPoint]: FluidObject = {
```

Comment:
This feels like a layering inversion, I particularly don't like it.

Reply:
currently it just works by convention. there isn't any enforcement. the same pattern could be followed, but in a more decoupled way, this is just one possible way to do it. adding something new made it apparent where changes are needed.

I don't think i'm familar with layering inversion is a principle, i know 
(Dependecy Inversion) [https://en.wikipedia.org/wiki/Dependency_inversion_principle], which is usually considered a good thing, but it can also lead to anti patterns like service location.